### PR TITLE
add inspected buildings notebook

### PR DESCRIPTION
**Pull request by apulverizer**
This adds a notebook showing how buildings that have been walked around can be detected. 

Todo:
- [x] wait for 1.8 of the ArcGIS API for Python to be released
- [x] get real track data near/around those buildings
- [x] fix missing map screenshots

### CODE REVIEW

File: notebooks/examples/Identify Inspected Buildings.ipynb [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "start_time = datetime(2020, 2, 1)\n",
```
Comment (niravi) [inline notebooks/examples/Identify Inspected Buildings.ipynb]: You could break out pendulum here if you wanted :)

`start_time = pendulum.datetime(2020, 2, 1, 0, 59, 59).format("YYYY-MM-DD HH:mm:ss", formatter='alternative')` (no formatter variable if using pendulum 2+)

File: notebooks/examples/Identify Inspected Buildings.ipynb [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    "end_time = datetime(2020, 2, 2)\n",
+    "where_clause = f\"location_timestamp >= timestamp '{start_time.strftime('%Y-%m-%d %H:%M:%S')}' AND location_timestamp < timestamp '{end_time.strftime('%Y-%m-%d %H:%M:%S')}'\"\n",
+    "tracks_layer = arcgis.features.FeatureLayer(f'{gis.properties.helperServices[\"locationTracking\"][\"url\"]}/0', gis)\n",
+    "tracks_layer.filter = where_clause\n",
+    "webmap.add_layer(tracks_layer)\n",
```
Comment (niravi) [inline notebooks/examples/Identify Inspected Buildings.ipynb]: Can we use a unique webmap object here, rather than the one we just used? Can run into this fun bug

flash screen

Comment (niravi) [reply notebooks/examples/Identify Inspected Buildings.ipynb]: Logged at: https://github.com/ArcGIS/geosaurus/issues/3205

File: notebooks/examples/Identify Inspected Buildings.ipynb [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "buildings_df[\"BUFFERED\"] = buildings_df[\"SHAPE\"].geom.buffer(15)\n",
+    "buildings_df.spatial.set_geometry(\"BUFFERED\")\n",
+    "buildings_df.spatial.sr = sr\n",
+    "webmap.add_layer(buildings_df)\n",
```
Comment (niravi) [inline notebooks/examples/Identify Inspected Buildings.ipynb]: Failed with error 
```
~/geosaurus/src/arcgis/widgets/_mapview.py in add_layer(self, item, options)
    861         if options is None:
    862             options = {}
--> 863         self._add_layer_to_widget(item, options)
    864 
    865     def _add_layer_to_webmap(self, item, options):

~/geosaurus/src/arcgis/widgets/_mapview.py in _add_layer_to_widget(self, item, options)
    908         elif isinstance(item, DataFrame):
    909             if hasattr(item, 'spatial'):
--> 910                 self.add_layer(item.spatial.to_featureset())
    911             else:
    912                 raise Exception('Could not add DataFrame to map it is not a spatially enabled DataFrame')

~/geosaurus/src/arcgis/features/geo/_accessor.py in to_featureset(self)
   2478         """
   2479         from arcgis.features import FeatureSet
-> 2480         return FeatureSet.from_dataframe(self._data)
   2481     #----------------------------------------------------------------------
   2482     def to_feature_collection(self,

~/geosaurus/src/arcgis/features/feature.py in from_dataframe(df)
    746         elif isinstance(df, pd.DataFrame) and \
    747              not df.spatial.name is None:
--> 748             fs = FeatureSet.from_dict(df.spatial.__feature_set__)
    749             df.set_index(old_idx,  inplace=True)
    750             return fs

~/geosaurus/src/arcgis/features/geo/_accessor.py in __feature_set__(self)
   2271             "globalIdFieldName" : "",
   2272             "displayFieldName" : "",
-> 2273             "geometryType" : _geom_types[type(self._data[self.name][self._data[self.name].first_valid_index()])],
   2274             "spatialReference" : sr,
   2275             "fields" : [],

~/anaconda3/envs/field-apps-web-automation/lib/python3.7/site-packages/pandas/core/series.py in __getitem__(self, key)
    869         key = com.apply_if_callable(key, self)
    870         try:
--> 871             result = self.index.get_value(self, key)
    872 
    873             if not is_scalar(result):

~/anaconda3/envs/field-apps-web-automation/lib/python3.7/site-packages/pandas/core/indexes/base.py in get_value(self, series, key)
   4386                 # try that
   4387                 try:
-> 4388                     iloc = self.get_loc(key)
   4389                     return s[iloc]
   4390                 except KeyError:

~/anaconda3/envs/field-apps-web-automation/lib/python3.7/site-packages/pandas/core/indexes/range.py in get_loc(self, key, method, tolerance)
    351             except ValueError:
    352                 raise KeyError(key)
--> 353         return super().get_loc(key, method=method, tolerance=tolerance)
    354 
    355     @Appender(_index_shared_docs["get_indexer"])

~/anaconda3/envs/field-apps-web-automation/lib/python3.7/site-packages/pandas/core/indexes/base.py in get_loc(self, key, method, tolerance)
   2646                 return self._engine.get_loc(key)
   2647             except KeyError:
-> 2648                 return self._engine.get_loc(self._maybe_cast_indexer(key))
   2649         indexer = self.get_indexer([key], method=method, tolerance=tolerance)
   2650         if indexer.ndim > 1 or indexer.size > 1:

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/index_class_helper.pxi in pandas._libs.index.Int64Engine._check_type()

KeyError: None
```

File: notebooks/examples/Identify Inspected Buildings.ipynb (hunk 2 above)
Comment (apulverizer) [reply notebooks/examples/Identify Inspected Buildings.ipynb]: Thanks for loggin that. Yeah this is annoying. I saw this and chose to ignore it for now. Probably will have to have separate webmap objects, or only show the webmap once or something.

Comment (niravi) [reply notebooks/examples/Identify Inspected Buildings.ipynb]: This has been resolved in the latest master

Comment (apulverizer) [reply notebooks/examples/Identify Inspected Buildings.ipynb]: Nice, I was using the same one, so that the map is updated and displayed as you progress through the notebook.

File: notebooks/examples/Identify Inspected Buildings.ipynb (hunk 3 above)
Comment (apulverizer) [reply notebooks/examples/Identify Inspected Buildings.ipynb]: Do you have shapely installed?

Comment (niravi) [reply notebooks/examples/Identify Inspected Buildings.ipynb]: I thought I did, but let me try again

File: notebooks/examples/Identify Inspected Buildings.ipynb [hunk 4]
```diff
@@ -0,0 +1,287 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Detect Inspected Buildings\n",
+    "\n",
+    "In damage assessment inspections, it's common that the structure needs to be fully inspected. One way to ensure that, is by having inspectors walk around the structure. In this example notebook, we'll how tracks can be used accomplish to identify structures that have been inspected. We'll use some sample data in Portland, ME to demonstrate this."
```
Comment (niravi) [inline notebooks/examples/Identify Inspected Buildings.ipynb]: Bad comma - either: `One way to ensure that is by having inspectors walk around the structure` or `One way to ensure that: having inspectors walk around the structure`

Comment (niravi) [reply notebooks/examples/Identify Inspected Buildings.ipynb]: Missing "see" in between "We'll" and "how"

Comment (niravi) [reply notebooks/examples/Identify Inspected Buildings.ipynb]: Remove "accomplish"

File: notebooks/examples/Identify Inspected Buildings.ipynb [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
...jsY7KUCK9sx8DAAPpZEgFX8j3Icyew9fDTwwZW2448nxWwG5tnY1fHS3BJp9+3f3vrNZx33rk4//zpmDP7jyX1PI+G7VTO9p+fehLnnz8dF154Id7922JN7/cN61fj7//+73HOOedg/qNzx/xMi15fiK9ccgmmTJmCP9z1W3Csrgog24FQ818QWnULEtLOCYHsQaYBUsON2PDJaAO02TiiW4F4PLkAFQz7sPXIcxnjaMPBBZBF7SfIAmfTzEMBKLy1o1pEEXo4+ic+f3iiZead2ERMQnY+amHC4MX8TNLcMo9+M10TiTMp9/2Jhm2Dw/OpqqI4y2omYTL0DE8aTgOuAAttAjPKtbRQV2ctVCrAnQmiCB1URfsFDkWjMk6vxwOfwsFhoQounbGwBIyGpOFZeR4meihccqfVxVvBkIYMd9R8TctIorR+bb0uM//7TBNNGvKufPHKHFRfcRFZo49oNApZ4BDJM987lwJuASLPla1UHAA8Xg+6ha60UvF2dTdMahdcKouQyiPqk+BTOMi8A4oiIxgMVgVcp45gMAhJ5OGRizvnqXx1LRYy4vE4eI6DeRRsO+w2TcvpT0P2aV+AqCpB5E9hT9craWXkaZDdch/2HXsLingaLt989SWcd965+Lu/+wKe+/OTaGpYrglkZ4PtJx6bh/POOw8XXfRZfPj+O5rd6xaWwPvvvY3p06fj85//HF5b+HLWz/LWm6/j0ku/irq6OvzPHbPgYPoqD9luO0K7n0Fo7e1IOPdMDGQTHyOw9HrIXQ1Yd2B+xjhZe2AeJDcDIOlqv/HQkxk/s639Bc0M0TJh265ZMgSQXISvRdimSQNsFm2yy6tR/YITW4hYxQGklrSdjMIuKnmfY7u1NhzJGcoIM01M+BhkONenZqHnLKs5CRn9zGl4dgtWMGRuFz2GMkK05efqrJVs5k+3C7nVTIIy6fLOiM5X0aEyTk0ivyRxeIGmEIdwXZGGZ4WIJvWwMqYxF2xIQj9mRndK+jxixnLJzJz5JeTJ822AT86dEzxSTsGBgYEBTSZ2oVAIiuAoOvbLIyfNxMpVKh6Px8HLHLrkg2hVt+KQuhMmtQtOlUFITZZcD7gEyLx9qO+6+vKuw+EwZEmEW3IgUmKWuU/RJgUBSFY12Gz9w88eu91WsOlePBFDJBZEKKIiEgum/VkgkAnZI+V3WUCwn2D/iS
```
Comment (niravi) [inline notebooks/examples/Identify Inspected Buildings.ipynb]: Remove number before this point and add colon after

Comment (niravi) [inline notebooks/examples/Identify Inspected Buildings.ipynb]: Apostrophe after users

Comment (niravi) [reply notebooks/examples/Identify Inspected Buildings.ipynb]: Or make 2/3 subpoints, similar to what you've done in step 4

Comment (niravi) [inline notebooks/examples/Identify Inspected Buildings.ipynb]: Add colon `For each user within the buffered area:`

Comment (niravi) [inline notebooks/examples/Identify Inspected Buildings.ipynb]: change to either `track points` or `tracks' points`

Comment (niravi) [inline notebooks/examples/Identify Inspected Buildings.ipynb]: Add colon at the end

Comment (niravi) [inline notebooks/examples/Identify Inspected Buildings.ipynb:177]: This is probably the most confusing part of the code, especially if you're not great w Spatial Dataframes. I'd include a comment on what you're doing here

Comment (niravi) [inline notebooks/examples/Identify Inspected Buildings.ipynb]: Colon at the end

Comment (niravi) [inline notebooks/examples/Identify Inspected Buildings.ipynb:259]: Shouldn't the two buildings between Commercial and Gold St also be inspected, based on your track data? Not seeing that in the webmap screenshot

Comment (apulverizer) [reply notebooks/examples/Identify Inspected Buildings.ipynb:177]: Yeah I agree, I added a comment

Comment (apulverizer) [reply notebooks/examples/Identify Inspected Buildings.ipynb:259]: I updated the description to explain why they were not considered inspected.

**Review by niravi — APPROVED** [review]
